torchtune/models/flamingo/_encoder.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional
+
+import torch
+
+from torch import nn, Tensor
+from torchtune.modules.model_fusion import register_fusion_module
+from torchtune.modules.transformer import _get_clones
+
+
+class FlamingoProjectionHead(nn.Module):
+    """Projection transformer to adapt the output of a
+    pretrained encoder to a pretrained decoder model.
+
+    Args:
+        layer (nn.Module): Transformer Decoder layer
+        num_layers (int): Number of Transformer Decoder layers
+        output (nn.Module): Output linear layer. Input dim is
+            (num_hidden + 1) * encoder_dim and output is decoder_dim.
+        num_hidden_inputs (int): Number of expected hidden state inputs
+    """
+
+    def __init__(
+        self,
+        layer: nn.Module,
+        num_layers: int,
+        output: nn.Module,
+        num_hidden_inputs: int = 0,
+    ) -> None:
+        super().__init__()
+        self.layers = _get_clones(layer, num_layers)
+        self.output = output
+        self.num_hidden = num_hidden_inputs
+
+    def forward(
+        self, x: Tensor, hidden_states: Optional[List[Tensor]] = None
+    ) -> Tensor:
+        """
+        Args:
+            x (Tensor): input tensor with shape [b x i x t x e x d]
```

Comment:
sorry this hurt my tiny brain
```suggestion
            x (Tensor): input tensor with shape [b * i * t * e * d]
```